main/modes/system/swsnCreator/swsnCreator.c

```diff
@@ -0,0 +1,1698 @@
+static void copyListToSona(swadgesona_t* swsn);
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+swadgeMode_t swsnCreatorMode = {
+    .modeName          = sonaModeName,
+    .wifiMode          = NO_WIFI,
+    .usesAccelerometer = true,
+    .fnEnterMode       = swsnEnterMode,
+    .fnExitMode        = swsnExitMode,
+    .fnMainLoop        = swsnLoop,
+};
+
+swsnCreatorData_t* scd;
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+static void swsnEnterMode(void)
+{
+    scd = (swsnCreatorData_t*)heap_caps_calloc(1, sizeof(swsnCreatorData_t), MALLOC_CAP_8BIT);
+
+    // Load
+    scd->tabSprs = heap_caps_calloc(ARRAY_SIZE(tabImages), sizeof(wsg_t), MALLOC_CAP_8BIT);
+    for (int idx = 0; idx < ARRAY_SIZE(tabImages); idx++)
+    {
+        loadWsg(tabImages[idx], &scd->tabSprs[idx], true);
+    }
+    loadFont(RODIN_EB_FONT, &scd->fnt, true);
+
+    // Load audio
+    loadMidiFile(ODE_MID, &scd->bgm, true);
+    scd->bgmPlayer       = globalMidiPlayerGet(MIDI_BGM);
+    scd->bgmPlayer->loop = true;
+    midiGmOn(scd->bgmPlayer);
+    globalMidiPlayerPlaySong(&scd->bgm, MIDI_BGM);
+
+    // If the SP swadgesona isn't saved yet, automatically load into creating the SP Sona
+    size_t len = sizeof(swadgesonaCore_t);
+    if (!readNvsBlob(spSonaNVSKey, &scd->activeSona.core, &len))
+    {
+        generateRandomSwadgesona(&scd->activeSona);
+        scd->activeSona.name = *getSystemUsername();
+        strcpy(scd->nickname, scd->activeSona.name.nameBuffer);
+        scd->slot  = 255; // Indicates it's the SP Sona
+        scd->state = CREATING;
+        readNvs32(cursorNVS, &scd->cursorType);
+        loadWsg(scd->cursorType, &scd->cursorImage, true);
+    }
+
+    swsnResetMenu();
+}
+
+static void swsnExitMode(void)
+{
+    globalMidiPlayerStop(true);
+    unloadMidiFile(&scd->bgm);
+    deinitMenuMegaRenderer(scd->renderer);
+    deinitMenu(scd->menu);
+    freeFont(&scd->fnt);
+    for (int idx = 0; idx < ARRAY_SIZE(tabImages); idx++)
+    {
+        freeWsg(&scd->tabSprs[idx]);
+    }
+    heap_caps_free(scd->tabSprs);
+    heap_caps_free(scd);
+}
+
+static void swsnLoop(int64_t elapsedUs)
+{
+    buttonEvt_t evt;
+    switch (scd->state)
+    {
+        case MENU:
+        {
+            while (checkButtonQueueWrapper(&evt))
+            {
+                scd->menu = menuButton(scd->menu, evt);
+            }
+            drawMenuMega(scd->menu, scd->renderer, elapsedUs);
+            break;
+        }
+        case VIEWING:
+        {
+            break;
+        }
+        case CREATING:
+        {
+            switch (scd->cState)
+            {
+                case BODY_PART:
+                {
+                    // Input
+                    runCreator(evt);
```

Comment:
CI failed here, and with reason. `evt` isn't initialized, nor is `checkButtonQueueWrapper()` called before calling `runCreator()`. I'll initialize `evt`, but you might want to get the input here.

I don't know why the OSX compiler is more stringent, but I'm glad that it is.

Reply:
Oh, `runCreator()` calls `checkButtonQueueWrapper()`, and likewise `panelOpen()` calls `panelInput()` calls `checkButtonQueueWrapper()`. So that's why it works at least.

So either check the button here and pass it as an argument, or don't pass it at all and declare it when `checkButtonQueueWrapper()` is actually called. I have a preference for the former.